test/conformance/runtime/readiness_probe_test.go

```diff
@@ -132,3 +142,98 @@ func TestProbeRuntime(t *testing.T) {
 		}
 	}
 }
+
+// This test validates the behaviour of readiness probes *after* initial startup.
+// The current behaviour is not ideal: when a pod goes unready after startup
+// and there are no other pods in the revision we hang, potentially forever,
+// which may not be what a user wants.
+// See https://github.com/knative/serving/issues/10765.
+func TestProbeRuntimeAfterStartup(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.Readiness,
+	}
+
+	test.EnsureTearDown(t, clients, &names)
+	resources, err := v1test.CreateServiceReady(t, clients, &names, v1opts.WithReadinessProbe(
+		&corev1.Probe{
+			// This behaviour is only the case where periodSeconds=0, because probes
+			// are ignored after startup when periodSeconds>0
+			// See https://github.com/knative/serving/issues/10764.
+			PeriodSeconds: 0,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path: "/healthz",
+				},
+			},
+		}))
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+
```

Comment:
I hate to have to bring this up but... Do we have to insert a `WaitForEndpointState` stanza here to make sure the service actually responds properly? (Ref #11404)

Reply:
ugh. hah. hm. yeah. :). This was initially part of the existing test that does that. Guess I'll copy that bit in here.

Reply:
(done)